tests/cpp/operator/mkldnn.cc

```diff
@@ -834,4 +855,89 @@ TEST(MKLDNN_BASE, MKLDNNSum) {
   }
 }
 
+TEST(MKLDNN_BASE, CreateMKLDNNMem) {
```

Comment:
How is this test different from the test on sum?
https://github.com/azai91/incubator-mxnet/blob/eabeb4330537369c19ec9266396844d798bfeeec/tests/cpp/operator/mkldnn.cc#L797

Reply:
test on sum doesn't use createmkldnnmem/commitoutput. I use the sum helper for this test cause copy doesn't support writeinplace (or doesnt really make sense). there is a separate mkldnnsum test that doesn't use createmkldnnmem/commitout that solely tests the purpose of the mkldnnsum. for future test on helper functions, they will follow a similar format to mkldnnsum without createmkldnn/commitout so it will be easier to see if the problem is with the helper or with createmkldnn/commitoutput.

Reply:
I'm not sure what you mean. The test on sum runs elemwise_add, which actually runs MKLDNNSumForward. MKLDNNSumForward calls CreateMKLDNNMem and CommitOutput. Do I miss something?

Reply:
oh you mean the test for the sum operator (vs the sum helper)? yes it does do that. the purpose of also having a special CreateMKLDNNMem/CommitOutput test is to test just those two functions. the sum operator test also tests all the steps from calling InvokeOp -> PushFCompute -> Fallback etc.

Reply:
it doesn't make the test of CreateMKLDNNMem/CommitOutput very explicit. could you please verify that all branches in CreateMKLDNNMem and CommitOutput are tested.

Reply:
kk. added kNoop so it tests all 4 requests types.